src/Playroom/Playroom.css.ts

```diff
@@ -27,71 +34,60 @@ globalStyle('body', {
   margin: 0,
 });
 
-export const root = sprinkles({
-  height: 'viewport',
-  width: 'viewport',
-});
-
-export const previewContainer = sprinkles({
-  position: 'absolute',
-  inset: 0,
-});
-
-export const editorSize = createVar();
-
-export const previewContainerPosition = styleVariants({
+export const root = styleVariants({
   right: {
-    right: `max(${editorSize}, ${MIN_WIDTH}px)`,
+    flexDirection: 'row',
   },
   bottom: {
-    bottom: `max(${editorSize}, ${MIN_HEIGHT}px)`,
+    flexDirection: 'column',
   },
   undocked: {},
 });
 
-export const resizableContainer = style([
+// Prevents the editor growing off screen
+// when resizable value increases past maximum size
+export const previewContainer = style({
+  minWidth: 0,
+  minHeight: 0,
+});
+
+export const editorSize = createVar();
+
+export const resizable = style([
   sprinkles({
-    bottom: 0,
-    right: 0,
     overflow: 'hidden',
     boxShadow: 'small',
-    transition: 'slow',
   }),
-  // @ts-expect-error Shouldnt need to but types do not like `!important`
-  {
-    position: 'absolute !important', // override re-resizable's inline style
-  },
 ]);
 
-export const resizableContainer_isHidden = style({});
-
-export const resizableContainer_isRight = style([
-  sprinkles({
-    top: 0,
-  }),
-  {
-    maxWidth: '90vw',
-    selectors: {
-      [`&${resizableContainer_isHidden}`]: {
-        transform: 'translateX(100%)',
-      },
-    },
+// override re-resizable's inline style
+// clamp ensures editor transitioning to correct size
+// before applying minimum and maximum sizes
+export const resizableSize = styleVariants({
+  right: {
+    width: `clamp(${[`${MIN_WIDTH}px`, editorSize, MAX_WIDTH].join(
+      ', '
+    )}) !important`,
```

Comment:
I feel like _not_ using `Array.join` here would look cleaner. Just comma-separated values.

```suggestion
  right: {
    width: `clamp(${MIN_WIDTH}px, ${editorSize}, ${MAX_WIDTH}) !important`,
```